cscs-checks/microbenchmarks/likwid/likwid.py

```diff
@@ -0,0 +1,176 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class BandwidthBase(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.build_system = 'Make'
+        self.build_system.flags_from_environ = False
+        self.sourcesdir = 'https://github.com/RRZE-HPC/likwid.git'
+        self.variables = {
+            'LD_LIBRARY_PATH': './lib:$LD_LIBRARY_PATH',
+            'PATH': './bin:./sbin:$PATH'
+        }
+
+        self.executable = 'bin/likwid-bench'
+
+        self.num_tasks = 1
+        self.num_tasks_per_core = 2
+        self.system_num_cpus = {
+            'daint:mc':  72,
+            'daint:gpu': 24,
+            'dom:mc':  72,
+            'dom:gpu': 24,
+        }
+        self.system_numa_domains = {
+            'daint:mc':  ['S0', 'S1'],
+            'daint:gpu': ['S0'],
+            'dom:mc':  ['S0', 'S1'],
+            'dom:gpu': ['S0'],
+        }
+
+        # Test each level at half capacity times nthreads per domain
+        self.system_cache_sizes = {
+            'daint:mc':  {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
+                          'memory': '1800MB'},
+            'daint:gpu': {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
+                          'memory': '1200MB'},
+            'dom:mc':    {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
+                          'memory': '1800MB'},
+            'dom:gpu':   {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
+                          'memory': '1200MB'},
+        }
+
+        self.maintainers = ['SK']
+        self.tags = {'diagnostic'}
+
+        bw_pattern = sn.extractsingle(r'MByte/s:\s*(?P<bw>\S+)',
+                                      self.stdout, 'bw',  float)
+
+        self.sanity_patterns = sn.assert_ge(bw_pattern, 0.0)
+        self.perf_patterns = {
+            'bandwidth': bw_pattern
+        }
+
+    def setup(self, partition, environ, **job_opts):
+        super().setup(partition, environ, **job_opts)
+
+        self.postbuild_cmd = ['make install PREFIX=%s INSTALL_CHOWN='
+                              '\'-g csstaff -o sebkelle\'' % self.stagedir]
```

Comment:
I would run with `srun -Cgpu,perf` instead

Reply:
Likwid can be build in different ways, see their [wiki](https://github.com/RRZE-HPC/likwid/wiki/likwid-accessD#enable-and-build).  
In production, likwid is build with `ACCESSMODE = perf_event`, see the [.eb](https://github.com/eth-cscs/production/blob/master/easybuild/easyconfigs/l/likwid/likwid-4.3.3-perf_event.eb#L37) which should be enough to avoid to have to call `chmod`.